src/dos/drive_local.cpp

```diff
@@ -375,21 +390,75 @@ bool localDrive::FindNext(DOS_DTA & dta) {
 	return true;
 }
 
-bool localDrive::GetFileAttr(char * name,Bit16u * attr) {
+bool localDrive::GetFileAttr(char *name, uint16_t *attr)
+{
 	char newname[CROSS_LEN];
 	safe_strcpy(newname, basedir);
 	safe_strcat(newname, name);
 	CROSS_FILENAME(newname);
 	dirCache.ExpandName(newname);
 
+#if defined(WIN32)
+	Bitu attribs = GetFileAttributes(newname);
+	if (attribs == INVALID_FILE_ATTRIBUTES) {
+		DOS_SetError((uint16_t)GetLastError());
+		return false;
+	}
+	*attr = attribs & 0x3f;
+	return true;
+#else
+	struct stat status;
+	if (stat(newname, &status) == 0) {
+		*attr = status.st_mode & S_IFDIR ? 0 : DOS_ATTR_ARCHIVE;
+		if (status.st_mode & S_IFDIR)
+			*attr |= DOS_ATTR_DIRECTORY;
+		if (!(status.st_mode & S_IWUSR))
+			*attr |= DOS_ATTR_READ_ONLY;
+		return true;
+	}
+	*attr = 0;
+	return false;
+#endif
+}
+
+bool localDrive::SetFileAttr(const char *name, uint16_t attr)
```

Comment:
`uint16_t attr` -> `const uint16_t attr`? (maybe)

Reply:
Done for all SetFileAttrih functions.